frontend/pluginloader.lua

```diff
@@ -85,14 +94,74 @@ function PluginLoader:loadPlugins()
     end
 
     -- set package path for all loaded plugins
-    for _,plugin in ipairs(self.plugins) do
+    for _,plugin in ipairs(self.enabled_plugins) do
         package.path = string.format("%s;%s/?.lua", package.path, plugin.path)
         package.cpath = string.format("%s;%s/lib/?.so", package.cpath, plugin.path)
     end
 
-    table.sort(self.plugins, function(v1,v2) return v1.path < v2.path end)
+    table.sort(self.enabled_plugins, function(v1,v2) return v1.path < v2.path end)
+
+    return self.enabled_plugins, self.disabled_plugins
+end
+
+function PluginLoader:genPluginManagerSubItem()
+    local enabled_plugins, disabled_plugins = {}, {}
+    if self.all_plugins == nil then
+        self.all_plugins = {}
+        enabled_plugins, disabled_plugins = self:loadPlugins()
+    end
+
+    for _, plugin in ipairs(enabled_plugins) do
+        local element = {}
+        element.fullname = plugin.fullname or plugin.name
+        element.name = plugin.name
+        element.description = plugin.description
+        element.enable = true
+        table.insert(self.all_plugins, element)
+    end
+
+    for _, plugin in ipairs(disabled_plugins) do
+        local element = {}
+        element.fullname = plugin.fullname or plugin.name
+        element.name = plugin.name
+        element.description = plugin.description
+        element.enable = false
+        if not OBSOLETE_PLUGINS[element.name] then
+            table.insert(self.all_plugins, element)
+        end
+    end
+    table.sort(self.all_plugins, function(v1, v2) return v1.fullname < v2.fullname end)
 
-    return self.plugins
+    local plugin_table = {}
+    for __, plugin in ipairs(self.all_plugins) do
+        table.insert(plugin_table, {
+            text = plugin.fullname,
+            checked_func = function()
+                return plugin.enable
+            end,
+            callback = function()
+                local InfoMessage = require("ui/widget/infomessage")
+                local UIManager = require("ui/uimanager")
+                local _ = require("gettext")
+                local plugins_disabled = G_reader_settings:readSetting("plugins_disabled") or {}
+                if plugin.enable then
+                    plugins_disabled[plugin.name] = plugin.enable
+                else
+                    plugins_disabled[plugin.name] = nil
+                end
+                plugin.enable = not plugin.enable
```

Comment:
I'd still like to see it done after the `plugin.enable = not plugin.enable`, so it uses the new value instead of the old one:
```lua
               plugin.enable = not plugin.enable
               if plugin.enable then
                   plugins_disabled[plugin.name] = nil
               else
                   plugins_disabled[plugin.name] = true
               end
```